byterun/caml/alloc.h

```diff
@@ -58,12 +70,15 @@ CAMLextern int caml_convert_flag_list (value, int *);
 /* Convenience functions to deal with unboxable types. */
 static inline value caml_alloc_unboxed (value arg) { return arg; }
 static inline value caml_alloc_boxed (value arg) {
-  value result = caml_alloc_small (1, 0);
-  Field (result, 0) = arg;
-  return result;
+  return caml_alloc_1(0, arg);
 }
 static inline value caml_field_unboxed (value arg) { return arg; }
-static inline value caml_field_boxed (value arg) { return Field (arg, 0); }
+static inline value caml_field_boxed (value arg) {
+  CAMLparam1(arg);
+  CAMLlocal1(v);
+  caml_read_field(arg, 0, &v);
+  CAMLreturn (v);
+}
```

Comment:
The implementation of this function made me wonder the following.  At present it is the case that many uses of `CAMLparam`, `CAMLlocal`, etc. can be elided (and are elided in many extant C stubs).  Is it ever the case for multicore that these macros are mandatory when they would previously not have been?  If so I think we need to start documenting that right away.